events/start_handler.go

```diff
@@ -84,12 +89,24 @@ func setupResolvConf(container *docker.Container) error {
 		}
 
 		if strings.HasPrefix(text, "search") {
+			existingSearchList := strings.Split(text, " ")
+			domainsToBeAdded := []string{}
+			finalSearchList := []string{"search"}
 			for _, domain := range getDNSSearch(container) {
 				if strings.Contains(text, " "+domain) {
 					continue
 				}
-				text = text + " " + domain
+				domainsToBeAdded = append(domainsToBeAdded, domain)
+			}
+			if value, ok := container.Config.Labels[RancherDNSPriority]; ok && value == "service_last" {
```

Comment:
I think you can just do
```go
if container.Config.Labels[RancherDNSPriority] == "service_last" {
  text = text + " " + strings.Join(domainsToBeAdded, " ")
} else {
  text = text.ReplaceN(text, "search", " " + strings.Join(domainsToBeAdded, " ") + " ", 1)
}
```
Then you don't need to actually parse anything.  I would guess the current code would fail if the resolv.conf was full of tabs.